exec/rpc/src/main/java/org/apache/drill/exec/rpc/ProtobufLengthDecoder.java

```diff
@@ -82,15 +79,7 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
         } else {
           // need to make buffer copy, otherwise netty will try to refill this buffer if we move the readerIndex forward...
           // TODO: Can we avoid this copy?
-          ByteBuf outBuf;
-          try {
-            outBuf = allocator.buffer(length);
-          } catch (OutOfMemoryException e) {
-            logger.warn("Failure allocating buffer on incoming stream due to memory limits.  Current Allocation: {}.", allocator.getAllocatedMemory());
-            in.resetReaderIndex();
-            outOfMemoryHandler.handle();
-            return;
-          }
+          ByteBuf outBuf = allocator.buffer(length);
```

Comment:
So I was looking at how we are using `ProtobufLengthDecoder` in our code, and we use in the control, server and user tunnels both on the client and server side.
If I am not mistaken, the only derived class from ProtobufLengthDecoder that actually uses an rpc allocator is UserProtobufLengthDecoder and only on the server side of the user tunnel. All other implementations use the root allocator.
This may actually explain why I wasn't able to reproduce this issue no matter how much I reduced the "rpc:bit-data" max allocation.

am I missing something ?